### PR TITLE
Add support for diufferent NTS settings

### PR DESCRIPTION
**Pull request by Heidistein**

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -266,6 +284,14 @@
   Boolean $rtconutc                                                = false,
   Variant[Hash,Array[String]] $hwtimestamps                        = [],
   Optional[Stdlib::Unixpath] $dumpdir                              = undef,
+  Optional[String]  $ntsserverkey                                  = undef,
```
Comment (bastelfreak) [inline manifests/init.pp:287]: can you make the types more strict?
```suggestion
  Optional[Stdlib::Absolutepath] $ntsserverkey                     = undef,
```

**Comment by Heidistein** [comment]
Sorry about the shitload of commits. Tests are not my hobby

**Comment by bastelfreak** [comment]
@Heidistein thanks for the PR. can you please rebase and squash the commits into one?

**Comment by smortex** [comment]
In order to combine your commits, from your working directory:

```
git rebase -i origin/master # Interactively rewrite history
```

This will bring your editor listing all commits in your branch.  Keep the first line unchanged, and replace `pick` with `squash` for all subsequent lines.  Save and quit, your editor will pop again and allow you to combine all the commit messages in a single one.

Then push your updated branch:

```
git push -f            # Send the changes (-f is required because we re-wrote history)
```

**Comment by Heidistein** [comment]
Aight. git went heywire on me. It's a burning pile right now. I'll close this and redo it from scratch.

**Pull request closed without merging**